br: disable restore checksum (#20847)
pingcap/docs: br/br-snapshot-manual.md

> **Note:**
>

> > - The BR tool already supports self-adapting to GC. It automatically registers `backupTS` (the latest PD timestamp by default) to PD's `safePoint` to ensure that TiDB's GC Safe Point does not move forward during the backup, thus avoiding manually setting GC configurations.

Reviewer: ```suggestion
> The BR tool already supports self-adapting to GC. It automatically registers `backupTS` (the latest PD timestamp by default) to PD's `safePoint` to ensure that TiDB's GC Safe Point does not move forward during the backup, thus avoiding manually setting GC configurations.
```

- `--ratelimit`: The maximum speed **per TiKV** performing restore tasks. The unit is in MiB/s.
- `--log-file`: The target file where the `br` log is written.

> During restore, a progress bar is displayed in the terminal as shown in the following example. When the progress bar advances to 100%, the restore task is completed. After the restoration is complete, if table-level [checksum](#checksum) is enabled, the BR tool performs table data verification to ensure the logical integrity of the data. Note that file-level checksums are always performed to ensure the basic integrity of the restored files.

Reviewer 2: ```suggestion
During restore, a progress bar is displayed in the terminal as shown in the following example. When the progress bar advances to 100%, the restore task is completed. After the restore is complete, if table-level [checksum](#checksum) is enabled, the BR tool performs data verification on the table to ensure logical integrity of the data. Note that file-level checksums are always performed to ensure the basic integrity of the restored files.
```

    --crypter.method aes128-ctr \
    --crypter.key 0123456789abcdef0123456789abcdef
```

## Checksum

Checksum is a method used by the BR tool to verify the integrity of backup and restore data. BR supports two levels of checksums:

1. **File-level checksum**: Verifies the backup files themselves to ensure integrity during storage and transmission. This checksum is always enabled and cannot be disabled.

> 2. **Table-level checksum**: Verifies the integrity of table data content and confirms the business logic consistency of the data. This checksum is disabled by default but can be enabled through parameters.

Reviewer 3: ```suggestion
2. **Table-level checksum**: Verifies the integrity of table data content and confirms the business logic consistency of the data. You can enable or disable this checksum through parameters.
```

Author: hmm, I feel like it will not be clear to customer that we disable it by default and they might think it's still enabled?

Balancing performance and security considerations, BR handles table-level checksums as follows:

### Backup checksum

Starting from v8.5.0, when performing full backups, the BR tool does not calculate table-level checksums by default (`--checksum=false`) to improve backup performance. If you need to calculate table-level checksums during backup, you can explicitly specify `--checksum=true`. File-level checksums will always be calculated to ensure the integrity of backup files.

> Calculating table-level checksums can verify data integrity during backup but increases backup time. In most cases, it's safe to use the default setting (no table-level checksum) to improve backup speed.

Reviewer: ```suggestion
Performing table-level checksums can verify data integrity during backup but increases backup time. In most cases, it is safe to use the default setting (no table-level checksum) to improve backup speed.
```

### Restore Checksum

> Starting from v9.5.0, the BR tool does not perform table-level checksum verification (`--checksum=false`) by default during restore operations to improve restore performance. If you need to perform table-level checksum verification, you can explicitly specify `--checksum=true`. File-level checksum verification is always performed to ensure the basic integrity of restored data.

Reviewer 3: ```suggestion
Starting from v9.0.0, the BR tool does not perform table-level checksum verification (`--checksum=false`) by default during restore operations to improve restore performance. If you need to perform table-level checksum verification, you can explicitly specify `--checksum=true`. File-level checksum verification is always performed to ensure the basic integrity of restored data.
```

> After restoration, data validation is usually performed to ensure data security. When table-level checksums are disabled, the comprehensive validation step for table data is skipped, thereby accelerating the restore process. For scenarios with strict data integrity requirements, you may choose to enable table-level checksums.

Reviewer: ```suggestion
After restoration, data validation is usually performed to ensure data security. When table-level checksums are disabled, the comprehensive validation step for table data is skipped, thereby accelerating the restore process. For scenarios with strict data integrity requirements, you can enable table-level checksums.
```

> ### Checksum Configuration Examples

Reviewer: ```suggestion
### Checksum configuration examples
```

    --crypter.method aes128-ctr \
    --crypter.key 0123456789abcdef0123456789abcdef
```

## Checksum

Checksum is a method used by the BR tool to verify the integrity of backup and restore data. BR supports two levels of checksums:

1. **File-level checksum**: Verifies the backup files themselves to ensure integrity during storage and transmission. This checksum is always enabled and cannot be disabled.

> 2. **Table-level checksum**: Verifies the integrity of table data content and confirms the business logic consistency of the data. This checksum is disabled by default but can be enabled through parameters.

Reviewer 2: ```suggestion
2. **Table-level checksum**: verifies the integrity of table data and ensures the business logic consistency of the data. This level of checksum is disabled by default, and you can enable it using a parameter.
```

> Balancing performance and security considerations, BR handles table-level checksums as follows:

Reviewer: ```suggestion
The following sections describe how BR handles table-level checksums, balancing performance and data safety considerations.
```
